src/remote.c

```diff
@@ -31,8 +31,11 @@ static int add_refspec_to(git_vector *vector, const char *string, bool is_fetch)
 	GITERR_CHECK_ALLOC(spec);
 
 	if (git_refspec__parse(spec, string, is_fetch) < 0) {
+                giterr_set(
+                        GITERR_INVALID,
+                        "'%s' is not a valid refspec.", string ? string : "(null)");
 		git__free(spec);
-		return -1;
+		return GIT_EINVALIDSPEC;
```

Comment:
Generally, I like to see us report the error at the place that the error occurs.  In this case, I think that it's actually `git_refpsec__parse` that's deficient.  It will [return `-1`](https://github.com/libgit2/libgit2/blob/master/src/refspec.c#L136) on error, but never sets the error message.

The consumers of this function (including this one) expect that the error message has been set.  (There are other uses even in the same file.)

So if you wouldn't mind moving this `giterr_set` into `git_refspec__parse`, I think that probably makes a bit more sense.